cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/InMemoryDatasetFramework.java

```diff
@@ -246,7 +248,8 @@ private void resetRegistry() {
     registry = registryFactory.create();
     for (Map.Entry<String, ? extends DatasetModule> entry : defaultModules.entrySet()) {
       LOG.info("Adding Default module: " + entry.getKey() + " " + this.toString());
-      add(entry.getKey(), entry.getValue(), true);
+      // default modules in system namespace
+      add(Id.DatasetModule.from(Constants.SYSTEM_NAMESPACE, entry.getKey()), entry.getValue(), true);
```

Comment:
I did that because ``defaultModules`` cannot have a namespace other than ``SYSTEM_NAMESPACE``. If I do not hardcode it here and make it be a map from id to module, I'd have to construct the id everywhere (``DatasetServiceRuntimeModule``, ``DatasetTypeManager``, tests, etc).

Thoughts?